Fix polaroid post (#53)
Bootstragram/bootstragram-blog: _posts/2021-03-07-css-polaroid-effect-with-fluid-grids.md

---
layout: post
id: 6DE45545-CDE9-451F-B421-A1CD36888ED1
title: A CSS Polaroid Effect With Flexbox
author: Mick F
excerpt: >-
  Revisiting how to show images as if they were polardoids, in CSS, with
  technologies from 2021.
category: Journaling
tags:
  - CSS
  - Sass
  - Compass
  - Jekyll
revisions:
  "2010-09-14": Original post using Compass to author SASS.
  "2021-02-17": Updated with contemporary techniques.
---

Back in 2010, after reading [Zurb's Polaroids with CSS3 blog post][2], I told
myself it could be a really cool effect to present Dead Rooster's MP3

> selections[^1] so I adapted the code, made it work with then very hot'n'sexy

Reviewer: ```suggestion
selections[^1] so I adapted the code and made it work with the (at the time) very hot'n'sexy
```

_fluid grids_.

Flash forward to 2021.

I'm migrating Dead Rooster to modern tools. Let's revisit this effect to
re-enable it.

## Jekyll is a decent Sass/SCSS authoring tool

The original version of this post mentioned [**Compass**][3] as a good Sass
authoring framework. It is no longer actively maintainted and nowadays, I rarely

> author any CSS outside of a Jekyll project or of CSS-in-JS tools suchs as

Reviewer: ```suggestion
author any CSS outside of a Jekyll project or CSS-in-JS tools such as
```

[styled-components][4] or [Emotion][5].

We will use [Jekyll's Sass/SCSS support][6] today.

## The Polaroid Effect

<ul class="polaroids">
{% for item in site.data.playlist %}
  <li class="polaroid" href="{{ item.link | default: "#" }}" title="{{ item.artist }}">
    <img src="https://picsum.photos/seed/{{ item.artist }}/200" />
  </li>
{% endfor %}
</ul>

## How it works

> I created 2 codepens to compare the code from 2010 to the one of 2021:

Reviewer: ```suggestion
I created 2 codepens to compare the code from 2010 to the code of 2021:
```

- [Polaroid Effects 2010](https://codepen.io/dirtyhenry/pen/BaQPoRL)
- [Polaroid Effects 2021](https://codepen.io/dirtyhenry/pen/vYyaNWE)

## What changed since 2010?

> The old code was using a lot of fixed sizes and a lot of maths, with a

Reviewer: ```suggestion
The old code used a lot of fixed sizes and math, with a
```

> configurable variable number of columns at compile time that became a constant

Author: This one is actually intentional: variable vs constant.

at runtime. To mimic this behavior, we could have used a [**CSS Grid

> Layout**][8] since, on Dead Rooster, each post using this effect was featuring a

Reviewer: ```suggestion
Layout**][8] since, on Dead Rooster, each post using this effect featured a
```

> 3 ⨉ 3 grid of images. But on smaller screens, this was leading images to become

Reviewer: ```suggestion
3 ⨉ 3 grid of images. But on smaller screens, the code made the images
```
